mitodl/open-discussions
# Checksums for content archives and resources

Based on https://github.com/mitodl/open-discussions/pull/3923 which should be reviewed & merged first

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #3930 

#### What's this PR do?
- Calculates and saves checksums of edx content archives and ContentFiles within the archives.  Does not process them further if checksums match.

#### How should this be manually tested?
This shows how to test for MITx archives, but you can do MITx Online or xPro if you prefer, just swap out the relevant .env variables and management command.

- Use the RC values for `EDX_*`  settings in your .env
- If you haven't done so already, run `backpopulate_edx_data`
- Run `backpopulate_edx_files`.  It will take awhile, note the total runtime when finished.
- Run it again.  This time it should be about 10x faster.


#### Any background context you want to provide?
For content archives, my initial idea was to use the S3 `object.size` property as the comparison metric to determine if anything had changed, which would avoid having to actually download the archive to figure out the checksum.  But since there's a chance, however small, that a changes in the archive might not lead to a change in size, I decided to go with a real checksum instead.  However i can be easily persuaded to go back to size instead of checksum by whoever reviews this. 😄
